include/aws/io/pem.h

```diff
@@ -0,0 +1,95 @@
+#ifndef AWS_IO_PEM_READER_H
+#define AWS_IO_PEM_READER_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/io/io.h>
+
+AWS_EXTERN_C_BEGIN
+
+enum aws_pem_object_type {
+    AWS_PEM_TYPE_UNKNOWN = 0,
+    AWS_PEM_TYPE_X509_OLD,
+    AWS_PEM_TYPE_X509,
+    AWS_PEM_TYPE_X509_TRUSTED,
+    AWS_PEM_TYPE_X509_REQ_OLD,
+    AWS_PEM_TYPE_X509_REQ,
+    AWS_PEM_TYPE_X509_CRL,
+    AWS_PEM_TYPE_EVP_PKEY,
+    AWS_PEM_TYPE_PUBLIC_PKCS8,
+    AWS_PEM_TYPE_PRIVATE_RSA_PKCS1,
+    AWS_PEM_TYPE_PUBLIC_RSA_PKCS1,
+    AWS_PEM_TYPE_PRIVATE_DSA_PKCS1,
+    AWS_PEM_TYPE_PUBLIC_DSA_PKCS1,
+    AWS_PEM_TYPE_PKCS7,
+    AWS_PEM_TYPE_PKCS7_SIGNED_DATA,
+    AWS_PEM_TYPE_PRIVATE_PKCS8_ENCRYPTED,
+    AWS_PEM_TYPE_PRIVATE_PKCS8,
+    AWS_PEM_TYPE_DH_PARAMETERS,
+    AWS_PEM_TYPE_DH_PARAMETERS_X942,
+    AWS_PEM_TYPE_SSL_SESSION_PARAMETERS,
+    AWS_PEM_TYPE_DSA_PARAMETERS,
+    AWS_PEM_TYPE_ECDSA_PUBLIC,
+    AWS_PEM_TYPE_EC_PARAMETERS,
+    AWS_PEM_TYPE_EC_PRIVATE,
+    AWS_PEM_TYPE_PARAMETERS,
+    AWS_PEM_TYPE_CMS,
+    AWS_PEM_TYPE_SM2_PARAMETERS
+};
+
+/*
+ * Describes PEM object decoded from file.
+ * data points to raw data bytes of object (decoding will do additional base 64
+ * decoding for each object).
+ * type will be set to object type or to AWS_PEM_TYPE_UNKNOWN if it could not
+ * figure out type.
+ * type_buf are the types bytes, i.e. the string between -----BEGIN and -----
+ */
+struct aws_pem_object {
+    enum aws_pem_object_type type;
+    struct aws_byte_buf type_buf;
```

Comment:
trivial: It's a string, so why not save it as a string
```suggestion
    struct aws_string *type_str;
```